samples/api/oit_linked_lists/oit_linked_lists.cpp

```diff
@@ -0,0 +1,482 @@
+/* Copyright (c) 2023, Google
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "oit_linked_lists.h"
+
+OITLinkedLists::OITLinkedLists()
+{
+}
+
+OITLinkedLists::~OITLinkedLists()
+{
+	if (!device)
+	{
+		return;
+	}
+
+	vkDestroyPipeline(get_device().get_handle(), combine_pipeline, nullptr);
+	vkDestroyPipeline(get_device().get_handle(), gather_pipeline, nullptr);
+	vkDestroyPipelineLayout(get_device().get_handle(), pipeline_layout, nullptr);
+
+	vkDestroyDescriptorPool(get_device().get_handle(), descriptor_pool, VK_NULL_HANDLE);
+	vkDestroyDescriptorSetLayout(get_device().get_handle(), descriptor_set_layout, nullptr);
+
+	destroy_sized_objects();
+
+	instance_data.reset();
+	scene_constants.reset();
+
+	object.reset();
+}
+
+bool OITLinkedLists::prepare(const vkb::ApplicationOptions &options)
+{
+	if (!ApiVulkanSample::prepare(options))
+	{
+		return false;
+	}
+
+	camera.type = vkb::CameraType::LookAt;
+	camera.set_position({0.0f, 0.0f, -4.0f});
+	camera.set_rotation({0.0f, 0.0f, 0.0f});
+	camera.set_perspective(60.0f, static_cast<float>(width) / static_cast<float>(height), 256.0f, 0.1f);
+
+	load_assets();
+	create_constant_buffers();
+	create_descriptors();
+	create_sized_objects(width, height);
+	create_pipelines();
+
+	update_scene_constants();
+	fill_instance_data();
+	update_descriptors();
+	build_command_buffers();
+
+	prepared = true;
+	return true;
+}
+
+bool OITLinkedLists::resize(const uint32_t width, const uint32_t height)
+{
+	destroy_sized_objects();
+	create_sized_objects(width, height);
+	update_descriptors();
+	ApiVulkanSample::resize(width, height);
+	return true;
+}
+
+void OITLinkedLists::render(float delta_time)
+{
+	if (!prepared)
+	{
+		return;
+	}
+
```

Comment:
See below for my "Rotate" checkbox comment.

You could do something as simple as:
```
static float rot = 0.0f;
camera.type = vkb::CameraType::LookAt;
camera.set_rotation({rot, rot, 0.0f});
if (rotate)
	rot += delta_time * 5.0f;
```


Probably better with `rot` as a member variable though.

Reply:
I added a call to camera.rotate when the option is enabled.